velero-csi-plugin/csi_snapshotter.go

```diff
@@ -117,6 +117,14 @@ func (p *CSISnapshotter) Execute(item runtime.Unstructured, backup *velerov1api.
 	}
 	p.log.Infof("volumesnapshot class=%s", snapshotClass.Name)
 
+	// If deletetionPolicy is not Retain, then in the event of a disaster, the namespace is lost with the volumesnapshot object in it,
+	// the underlying volumesnapshotcontent and the volume snapshot in the storage provider is also deleted.
+	// In such a scenario, the backup objects will be useless as the snapshot handle itself will not be valid.
+	// For that reason, skip backup and error out
+	if snapshotClass.DeletionPolicy != snapshotv1beta1api.VolumeSnapshotContentRetain {
```

Comment:
I think for at least the beta period, this should probably be a warning, and if it becomes an issue we can move towards completely disabling snapshots. Originally, I wanted to make this policy a prerequisite, but I think that can be a documentation-only prereq.

Also, updating the deletion policy is unfortunately not possibly, or wasn't in the alpha. The `VolumeSnapshotClass` objects couldn't be patched at all after creation.

Reply:
Updated to warnings instead of errors